### PR TITLE
DOC: special: complete hypergeometric functions documentation

### LINKED ISSUE #6966: Hypergeometric Functions documentation is lacking
The hypergeometric functions `scipy.special.hyp1f1(a, b, x)`, `scipy.special.hyp2f1(a, b, x)`, are very poorly documented [here](https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.hyp1f1.html#scipy.special.hyp1f1).

Specifically, all the `hypxfx` functions don't take complex arguments and don't cite the algorithm they are using.

### PR DESCRIPTION
**Pull request by person142**
#### Reference issue

Closes https://github.com/scipy/scipy/issues/6966.

#### What does this implement/fix?

Make sure the docs for all the hypergeometric functions have

- Parameters/Returns
- A definition and reference
- Examples

### CODE REVIEW

File: scipy/special/add_newdocs.py [hunk 1]
```diff
@@ -3376,19 +3376,23 @@ def add_newdoc(name, doc):
 
 add_newdoc("hyp0f1",
     r"""
-    hyp0f1(v, x)
+    hyp0f1(v, x, out=None)
```
Comment (tylerjereddy) [inline scipy/special/add_newdocs.py]: It looks like the parameter named `x` in this line is designated as `z` in the docstring, including the original function / summation definition.

**Review by tylerjereddy — CHANGES_REQUESTED** [review]
Just tried to proofread a little as non-expert--only noticed one thing that might need a small fix?

File: scipy/special/add_newdocs.py (hunk 1 above)
Comment (person142) [reply scipy/special/add_newdocs.py]: Good catch; fixed it.

**Comment by ev-br** [comment]
This is great, thanks Josh

**Pull request merged**